charts/linkerd2/templates/sp-validator.yaml

```diff
@@ -41,8 +41,6 @@ spec:
   selector:
     matchLabels:
       {{.ControllerComponentLabel}}: sp-validator
-      {{.ControllerNamespaceLabel}}: {{.Namespace}}
```

Comment:
Service profile validator already has its own selectors in 2.4. Since these selectors are immutable, adding new match labels here will cause upgrade to fail. Hence, I have to revert this change. The same revert is also added to the proxy injector template.